### ISSUE TITLE
SIM102 Warning when using multiple if-statements with walrus operator

### ISSUE THREAD

**Issue by macintacos** [astral-sh/ruff#5471]
I have some nested walrus operator code that is making this warning show up. Code looks something like this:

```python
if foo := bar():
	if baz := buz():
		# use 'foo' and 'baz' here
```

There is no way for me to combine these two if statements into one, since that's not (to my knowledge) supported in Python. Could this warning be silenced in these situations?

**Comment by charliermarsh** [comment]
Hmm -- does this work, or am I misunderstanding the code?

```python
def bar():
    return 1

def buz():
    return 2

if (foo := bar()) and (baz := buz()):
    print(foo, baz)  # 1 2
```

**Comment by macintacos** [comment]
Ah you're totally right - closing!

**Comment by tylerlaprade** [comment]
I also ran into this problem and had to search Google for this answer. Could we please add the example to the [documentation](https://beta.ruff.rs/docs/rules/collapsible-if/) for this rule?
